From cbe0ad7b432e59768b0c2e88aaec2ed9abbe6998 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Tue, 13 Aug 2019 18:49:07 +0800
Subject: [PATCH 138/138] octeontx2-af: Compatible with the old firmware
 version

In commit 64ca66734d05 ("octeontx2-af: Move to rvu_fwdata version 1."),
we have changed the RVU_FWDATA_VERSION to 1 to match with the latest
firmware version, but this will break on the board which is still using
the old version firmware. So add the support to be compatible with the
old version firmware.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.h b/drivers/net/ethernet/marvell/octeontx2/af/rvu.h
index 6cb284b8027c..4f94471437da 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.h
@@ -380,6 +380,7 @@ struct mbox_wq_info {
 struct rvu_fwdata {
 #define RVU_FWDATA_HEADER_MAGIC	0xCFDA	/*Custom Firmware Data*/
 #define RVU_FWDATA_VERSION	0x0001
+#define RVU_FWDATA_VERSION_0	0x0000
 	u32 header_magic;
 	u32 version;		/* version id */
 
@@ -521,7 +522,8 @@ static inline int is_afvf(u16 pcifunc)
 static inline bool is_rvu_fwdata_valid(struct rvu *rvu)
 {
 	return (rvu->fwdata->header_magic == RVU_FWDATA_HEADER_MAGIC) &&
-		(rvu->fwdata->version == RVU_FWDATA_VERSION);
+		(rvu->fwdata->version == RVU_FWDATA_VERSION ||
+		 rvu->fwdata->version == RVU_FWDATA_VERSION_0);
 }
 
 int rvu_alloc_bitmap(struct rsrc_bmap *rsrc);
-- 
2.17.1

